bgpd/bgp_zebra.c

```diff
@@ -397,12 +399,23 @@ static int bgp_interface_address_delete(int command, struct zclient *zclient,
 			   ifc->ifp->name, buf);
 	}
 
+	/* Update the router id if the interface address being deleted is
+	 * same as the bgp router id
+	 */
+	p = ifc->address;
+	if ((p->family == AF_INET) &&
+		IPV4_ADDR_SAME(&bgp->router_id, &p->u.prefix4))
+		update_routerid = true;
+
 	if (if_is_operative(ifc->ifp)) {
 		bgp_connected_delete(bgp, ifc);
 	}
 
 	connected_free(ifc);
 
+	if (update_routerid)
+		bgp_router_id_zebra_bump(vrf_id, NULL);
```

Comment:
You cannot pass `NULL` to `bgp_router_id_zebra_bump`. Did you even test this?

Also, as noted in my earlier comment, I think we should leave the router-id even if the address is deleted (but only if it is in use / bgp sessions are up.)

Reply:
(misread the patch.)